R/exposure_map_chart.R

```diff
@@ -0,0 +1,111 @@
+#' Create an interactive exposure map chart in an htmlwidget
+#'
+#' @param .data an exposure map data frame
+#' @param width width of exported htmlwidget in pixels (single integer value; default == NULL)
+#' @param height height of exported htmlwidget in pixels (single integer value; default == NULL)
+#' @param ... other options passed on to r2d3::r2d3() (see details)
+#'
+#' @description
+#' The `exposure_map_chart` function creates an interactive exposure map chart in an htmlwidget
+#'
+#' @md
+#' @export
+
+exposure_map_chart <-
+  function(.data, width = NULL, height = NULL, ...) {
+    .data <- export_data_utf8(.data)
+
+    dependencies <-
+      list(
+        system.file("js/map.js", package = "r2dii.interactive"),
+        system.file("js/d3-geo-projection.v2.min.js", package = "r2dii.interactive"),
+        system.file("js/d3-array.v2.min.js", package = "r2dii.interactive"),
+        system.file("js/world.js", package = "r2dii.interactive"),
+        system.file("js/text_dropdown_jiggle.js", package = "r2dii.interactive"),
+        system.file("css/2dii_gitbook_style.css", package = "r2dii.interactive"),
+        system.file("css/hide_styles.css", package = "r2dii.interactive")
+        )
+
+    op <- options(r2d3.shadow = FALSE)
+    on.exit(options(op), add = TRUE)
+
+    r2d3::r2d3(
+      data = .data,
+      script = system.file("render_exposure_map.js", package = "r2dii.interactive"),
+      dependencies = dependencies,
+      d3_version = 4,
+      width = width,
+      height = height,
+      container = "div",
+      ...
+    )
+  }
+
+
+#' Convert raw data into a exposure map data frame
+#'
+#' @param results_map results_map
+#' @param start_year start_year
+#' @param dataframe_translations dataframe_translations
+#' @param language_select two letter code for language (single string; default = "EN")
+#'
+#' @description
+#' The `as_exposure_map_data` function converts raw data into a exposure map data frame
+#'
+#' @import dplyr
```

Comment:
because of dplyr's heavy use of NSE and infix operators, plus my need for quite a few of dplyr's function, I find it easier and less messier to just import the whole package